src/coreclr/jit/compiler.h

```diff
@@ -9978,6 +9995,24 @@ class Compiler
 
         return false;
     }
+
+    //------------------------------------------------------------------------
+    // DoJitStressRex2Encoding- Answer the question: Do we force REX2 encoding.
+    //
+    // Returns:
+    //    `true` if user requests REX2 encoding.
+    //
+    bool DoJitStressRex2Encoding() const
+    {
+#ifdef DEBUG
+        if (JitConfig.JitStressRex2Encoding())
```

Comment:
Does this need to interplay with `JitStressEvexEncoding` or similar at all (or does that need to be renamed, to clarify its only SIMD EVEX?)

In particular, I'm considering what the behavior would be on a machine without EVEX but where stress REX2 is enabled (which can't exist for real hardware), so there may be some consideration of ensuring that EVEX is enabled when REX2 is being stressed so everything "lines up".

Reply:
> Does this need to interplay with `JitStressEvexEncoding` or similar at all (or does that need to be renamed, to clarify its only SIMD EVEX?)

We will use `JitStressPromotedEVEXEncoding` to distinguish between APX-EVEX and pre-APX EVEX.

> In particular, I'm considering what the behavior would be on a machine without EVEX but where stress REX2 is enabled (which can't exist for real hardware), so there may be some consideration of ensuring that EVEX is enabled when REX2 is being stressed so everything "lines up".

If we are simply stressing the "encodings", say letting all the compatible instructions to be encoded with the new encodings, since REX2 and APX-EVEX compatibility are tracked separately,  I think letting REX2 and APX-EVEX stand alone should be fine. 

But if we are to stress the register allocator to use the new registers (which seems to be another scenario.), we might need to ensure both encodings are available. 

Is this the case we are considering, or I have some misunderstanding?

Reply:
I'm basically trying to determine if the additional encoding is actually needed.

In reality you can have `SIMD_EVEX` or `SIMD_EVEX + APX_EVEX`, you can't have just `APX_EVEX`. Accordingly, the goal of the stress bit here is to catch any potential issues with the encoding, such as unhandled encodings; so the differentiation of `SIMD_EVEX` vs `GP_EVEX` from the stress perspective seems irrelevant.

Additionally, we're only going to be stressing whatever ISAs the JIT says are supported; so it would seem as though the existing `JitStressEvexEncoding` is already sufficient, which indicates that `SIMD_EVEX` is stressed. If the underlying JIT (or AltJit configuration) supports `APX`, then it will be stressed as well.

Reply:
Thanks for the elaboration, 

Indeed, `EVEX` can stand for both `SIMD_EVEX` and `APX_EVEX` and actually `APX_EVEX` related functionality requires machine already has `EVEX` support, either from Avx512 or Avx10 (so potentially we can create an implication that APX implies EVEX). Plus, how we handle instructions like `ADD` that have both `REX2` and `EVEX` under different stress modes may need to be considered. I'd like to discuss these topics more in the follow up PR for `APX_EVEX`.

And in this PR, according to what we expect from the encoding stress mode, I agree that stressing `REX2` may need `EVEX` stressed as well to avoid some unhandled EGPR incompatibility.